src/base.rs

```diff
@@ -112,7 +112,9 @@ pub fn compile_codegen_unit<'tcx>(tcx: TyCtxt<'tcx>, cgu_name: Symbol, supports_
         context.add_command_line_option("-mavx");
 
         for arg in &tcx.sess.opts.cg.llvm_args {
-            context.add_command_line_option(arg);
+            if !arg.starts_with("--x86-asm-syntax=") {
+                context.add_command_line_option(arg);
+            }
```

Comment:
This would break anyone that needs a specific assembly syntax in rustc's output, right?

Reply:
We're currently doing some tests.

This option is an LLVM option; for gcc it should be `-masm=intel`. I'm not sure what's the plan for supporting setting options for the different backends, but cg_gcc currently uses `llvm-args` to set gcc options. Since this option is to send arguments to the backend, I'm not sure it makes sense to do the mapping between the LLVM options and GCC options.

I guess we would need to discuss this somewhere to agree on the correct solution, though.